plugins/content_distribution/providers/cortex/lib/CortexApiDistributionEngine.php

```diff
@@ -407,8 +423,42 @@ protected function uploadVideo(KalturaCortexApiDistributionJobProviderData $apiD
 	{
 		try
 		{
-			$SystemIdentifier = $this->mediaUpload($apiDistributionJobProviderData->videoAssetFilePath);
+            $needDel = false;
+            $videoFilePath = $apiDistributionJobProviderData->videoAssetFilePath;
+            if (!$videoFilePath)
+                throw new KalturaException('No video asset to distribute, the job will fail',KalturaBatchJobAppErrors::BULK_ITEM_NOT_FOUND);
+
+            if (!kFile::checkFileExists($videoFilePath))
+                throw new KalturaDistributionException('The file ['.$videoFilePath.'] was not found (probably not synced yet), the job will retry');
+
+            list($isRemote, $remoteUrl) = kFile::resolveFilePath($videoFilePath);
+            $tempVideoFilePath = !$isRemote ? null : kFile::getExternalFile($remoteUrl, null, basename($videoFilePath));
+
+            if (FALSE === strstr($videoFilePath, "."))
+            {
+                $videoFilePathNew = $this->tempXmlPath . "/" . uniqid() . ".dme";
```

Comment:
i dont think that the dme makes sense here, its is part of the path for daily motion distribution but how it connects here?

Reply:
This logic was also in YoutubeAPI/DailyMotion, so Im not sure why is it necessary but I already showed this change to Yossi P. who approved this is the correct logic